### PR TITLE
Speed up for key pressed and image clicked

### PR DESCRIPTION
**Pull request by FunmiKesa**

### CODE REVIEW

File: selector_app.py [hunk 1]
```diff
@@ -611,9 +616,7 @@ def activate_deactivate_cells(n_rows, n_cols, n_left, n_right, n_up, n_down, n_k
     # Find the button that triggered this callback (if any)
     context = dash.callback_context
     if not context.triggered:
-        class_names = ['grouped-off focus' if i+j == 0 else 'grouped-off' for i in range(ROWS_MAX) for j in range(COLS_MAX)]
```
Comment (lukerm) [inline selector_app.py:614]: Good find!

File: selector_app.py [hunk 2]
```diff
@@ -548,7 +553,7 @@ def create_reactive_image_grid(n_row, n_col, image_list, image_data, image_path)
 
 
 @app.callback(
-    ALL_TD_ID_OUTPUTS + [Output('zoomed-image', 'children')],
+    ALL_TD_ID_OUTPUTS + [Output('zoomed-image', 'children'), Output('cell_last_clicked', 'data')],
```
Comment (lukerm) [inline selector_app.py:553]: Good idea to keep track of the last-clicked cell.

File: utils.py [hunk 3]
```diff
@@ -397,6 +397,95 @@ def image_cell_pressed(button_id, n_cols, image_list, *args):
...
+            assert 'grouped-on' in previous_class_clicked
+            assert 'focus' in previous_class_clicked
+            new_class_clicked = class_turn_off_keep_delete(class_toggle_grouped(class_toggle_focus(previous_class_clicked)))
+        cell_last_clicked = cell_loc
+        new_class_clicked = ' '.join(new_class_clicked)
+        new_classes[idx] = new_class_clicked
+
+
+    else:
+        cell_last_clicked = None
```
Comment (lukerm) [inline utils.py]: Does this `else` clause get used any more? It's the code from the old function, right?

File: utils.py [hunk 4]
```diff
@@ -442,6 +531,81 @@ def direction_key_pressed(button_id, n_rows, n_cols, image_list, *args):
...
+                        cell_last_clicked = [i, j]
+                    else:
+                        new_classes.append(my_class)
+
+    img_idx = cell_last_clicked[1] + cell_last_clicked[0]*n_cols
+    zoomed_img = html.Img(src=image_list[img_idx], style=config.IMG_STYLE_ZOOM) if len(image_list) > 0 else EMPTY_IMAGE
+    return new_classes, zoomed_img, cell_last_clicked
+
+
 def keep_delete_pressed(button_id, n_rows, n_cols, image_list, *args):
```
Comment (lukerm) [inline utils.py:434]: I think this may need to be updated, as the functionality behind these buttons doesn't work any more. I will have a think.

**Review by lukerm — CHANGES_REQUESTED** [review]
Hi Fumi! Thanks for looking at this. At first glance, navigation seems faster. But unfortunately the keep (s) / delete (d) shortcuts no longer work.

**Comment by lukerm** [comment]
Please also merge into the branch `feature/m.speed-up`.

File: utils.py (hunk 3 above)
Comment (FunmiKesa) [reply utils.py]: Yeah. I don't think it's needed again. Initially, I thought the initialization would be done in the else.

File: utils.py (hunk 4 above)
Comment (FunmiKesa) [reply utils.py:434]: Updated the code!

**Comment by FunmiKesa** [comment]
> Hi Fumi! Thanks for looking at this. At first glance, navigation seems faster. But unfortunately the keep (s) / delete (d) shortcuts no longer work.

I have updated the code for keeping and deleting.

**Review by lukerm — CHANGES_REQUESTED** [review]
Funmi, thanks for the changes - I'm happy to accept the PR! But first, just a few things to tidy up if you don't mind:

- In `utils.py`, remove the suffices "_speed_up" and delete the old functions that I wrote. 
  - Change all references to _speed_up functions in the `selector_app.py` script
- Remove any print statements / commented-out code that you have introduced.
- On [line 32](https://github.com/FunmiKesa/image-selector/blob/speed-up/selector_app.py#L32) of the main app, there is note about a known bug. Please now remove that note as your implementation fixed it :smile: 

If you could add a small number of comments or docstring about how the new functions work, that would be fab. But I'll leave it up to you if you want to do that!

**Review by lukerm — APPROVED** [review]
Looks good - thanks a lot!

**Pull request merged**